test/unit/async-config.size-limit.test.js

```diff
@@ -1,41 +1,33 @@
-import cheerio from 'cheerio'
-import fetch from 'node-fetch'
-import { getPort, loadFixture, Nuxt } from '../utils'
 
-let port
-let nuxt = null
-const url = route => 'http://localhost:' + port + route
-let responseSizes
+import { resolve } from 'path'
+import fs from 'fs-extra'
 
-describe('size-limit test', () => {
-  beforeAll(async () => {
-    const options = await loadFixture('async-config')
-    nuxt = new Nuxt(options)
-    await nuxt.ready()
+const distDir = resolve(__dirname, '../fixtures/async-config/.nuxt/dist')
 
-    port = await getPort()
-    await nuxt.server.listen(port, '0.0.0.0')
+const getResourcesSize = async (mode) => {
+  const manifestFile = resolve(distDir, 'server', `${mode}.manifest.json`)
+  const { all } = await import(manifestFile)
+  const resources = all.filter(filename => filename.endsWith('.js'))
+  let size = 0
+  for (const resource of resources) {
+    const stat = await fs.stat(resolve(distDir, 'client', resource))
+    size += stat.size / 1024
+  }
+  return size
+}
 
-    const { html } = await nuxt.server.renderRoute('/')
-    // Get all script URLs from the HTML
-    const $ = cheerio.load(html)
-    const scriptsUrls = $('script[src]')
-      .map((_, el) => $(el).attr('src'))
-      .get()
-      .map(url)
-    const resourceUrls = [url('/'), ...scriptsUrls]
-
-    // Fetch all resources and get their size (bytes)
-    responseSizes = await Promise.all(resourceUrls.map(async (url) => {
-      const response = await fetch(url).then(res => res.text())
-      return response.length
-    }))
+describe('nuxt basic resources size limit', () => {
+  it('should stay within the size limit range in legacy mode', async () => {
+    const LEGACY_JS_RESOURCES_KB_SIZE = 195
+    const legacyResourcesSize = await getResourcesSize('client')
+    expect(legacyResourcesSize).toBeLessThanOrEqual(LEGACY_JS_RESOURCES_KB_SIZE * 1.05)
+    expect(legacyResourcesSize).toBeGreaterThanOrEqual(LEGACY_JS_RESOURCES_KB_SIZE * 0.95)
   })
 
-  it('should stay within the size boundaries', () => {
-    const responseSizeBytes = responseSizes.reduce((bytes, responseLength) => bytes + responseLength, 0)
-    const responseSizeKilobytes = Math.ceil(responseSizeBytes / 1024)
-    // Without gzip!
```

Comment:
We should keep that piece of information somewhere around

Reply:
Do you mean gzip?

Reply:
I mean that it should be noted somewhere the we compare ungzipped file sizes :relaxed:

Reply:
It makes sense when we use http, but now it’s reading from fs directly, is it still necessary? Or we can add test for gzip or brotli?

Reply:
I see what you mean but from my experience, most people think of gzipped & minified numbers when it comes to benchmarking bundle sizes. So the comment wouldn't hurt here.

Gzip/brotli tests would help us as they would provide accurate base values for a basic nuxt application (without taking #6287 into consideration here)

Reply:
I’ll add compression size test